addons/purchase_stock/report/report_stock_rule.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+class ReportStockRule(models.AbstractModel):
+    _inherit = 'report.stock.report_stock_rule'
+
+    @api.model
+    def _get_rule_loc(self, rule, product_id):
```

Comment:
""" We override this method to handle `buy` rule which do not have a location_src_id.
"""